src/HDF5.jl

```diff
@@ -2007,18 +2010,41 @@ function hdf5_to_julia_eltype(objtype)
         T = HDF5Vlen{hdf5_to_julia_eltype(HDF5Datatype(super_id))}
     elseif class_id == H5T_COMPOUND
         N = Int(h5t_get_nmembers(objtype.id))
```

Comment:
```suggestion
        N = h5t_get_nmembers(objtype.id)
```
probably can do without the `Int` conversion just fine